File: capacity-club-api/src/domain-modules/address/model/filter/address.filter.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, IsOptional } from 'class-validator';

/**
 * Class used to filter address.
 * This class defines the criteria used to filter a list of address.
 */
export class AddressFilter {
  @ApiProperty()
  @IsOptional()
  @IsString()
  street: string;
  @ApiProperty()
  @IsOptional()
  @IsString()
  number: string;
  @ApiProperty()
  @IsOptional()
  @IsString()
  zipcode: string;
  @ApiProperty()
  @IsOptional()
  @IsString()
  town: string;
  @ApiProperty()
  @IsOptional()
  @IsString()
  country: string;
  @ApiProperty()
  @IsOptional()
  @IsString()
  complement: string;
}

File: capacity_club_mobile_app/lib/domain-features/data/address/model/filter/address_filter.dart
import 'package:capacity_club_mobile_app/core/model/extension/json_map_extension.dart';

class AddressFilter {
  final String? street;
  final String? number;
  final String? zipcode;
  final String? town;
  final String? country;
  final String? complement;

  AddressFilter({
    this.street,
    this.number,
    this.zipcode,
    this.town,
    this.country,
    this.complement,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{};

    json.addIfNotNull('street', street);
    json.addIfNotNull('number', number);
    json.addIfNotNull('zipcode', zipcode);
    json.addIfNotNull('town', town);
    json.addIfNotNull('country', country);
    json.addIfNotNull('complement', complement);

    return json;
  }
}
